qiskit/backends/ibmq/ibmqjob.py

```diff
@@ -0,0 +1,313 @@
+from qiskit._qiskiterror import QISKitError
+from qiskit._result import Result
+from qiskit._resulterror import ResultError
+
+logger = logging.getLogger(__name__)
+
+
+class IBMQJob(BaseJob):
+    """IBM Q Job class"""
+
+    def __init__(self, q_job, api, submit_info, is_device):
+        """IBMQJob init function.
+
+        Args:
+            q_job (QuantumJob): job description
+            api (IBMQuantumExperience): IBM Q API
+            submit_info (dict): this is the dictionary that comes back when
+                submitting a job using the IBMQuantumExperience API.
+            is_device (bool): whether backend is a real device  # TODO: remove this after Qobj
+        """
+        super().__init__()
+        self._q_job = q_job
+        self._api = api
+        self._submit_info = submit_info
+        self._backend_name = submit_info.get('backend', None).get('name', None)
+        self._job_id = submit_info['id']
+        self._status = JobStatus.QUEUED
+        self._status_msg = None
+        self._cancelled = False
+        self._exception = None
+        self._is_device = is_device
+
+    def result(self, timeout=None, wait=5):
+        # pylint: disable=arguments-differ
+        this_result = self._wait_for_job(timeout=timeout, wait=wait)
+        if self._is_device and self.done:
+            _reorder_bits(this_result)  # TODO: remove this after Qobj
+        return this_result
+
+    def cancel(self):
+        """Attempt to cancel job. Currently this is only possible on
+        commercial systems.
+
+        Returns:
+            bool: True if job can be cancelled, else False.
+
+        Raises:
+            QISKitError: if server returned error
+        """
+        if self._is_commercial:
+            hub = self._api.config['hub']
+            group = self._api.config['group']
+            project = self._api.config['project']
+            response = self._api.cancel_job(self._job_id, hub, group, project)
+            if 'error' in response:
+                err_msg = response.get('error', '')
+                self._exception = QISKitError('Error cancelling job: %s' % err_msg)
+                raise QISKitError('Error canceelling job: %s' % err_msg)
+            else:
+                self._cancelled = True
+                return True
+        else:
+            return False
+
+    @property
+    def status(self):
+        job_result = self._api.get_job(self._job_id)
+        stats = {'job_id': self._job_id}
+        _status = None
+        _status_msg = None
+        if 'status' not in job_result:
+            self._exception = QISKitError("get_job didn't return status: %s" %
+                                          (pprint.pformat(job_result)))
+            raise QISKitError("get_job didn't return status: %s" %
+                              (pprint.pformat(job_result)))
+        elif job_result['status'] == 'RUNNING':
+            # we may have some other information here
+            if 'infoQueue' in job_result:
```

Comment:
I think the fix to my comment above is just to add 
`_status = JobStatus.RUNNING` in the `if` here. Right now it is only in the `else`.

Reply:
Yes - that seems to be a needed fix! :+1: